server/src/main/java/com/github/animelist/animelist/model/profilepage/Block.java

```diff
@@ -0,0 +1,52 @@
+package com.github.animelist.animelist.model.profilepage;
+
+public abstract class Block {
+    private Width width;
+    private BlockType type;
+
+    public Block(Width width, BlockType type) {
+        this.width = width;
+        this.type = type;
+    }
+
+    public void setWidth(Width width) {
+        this.width = width;
+    }
+
+    public void setType(BlockType type) {
+        this.type = type;
+    }
+
+    public Width getWidth() {
+        return width;
+    }
+
+    public BlockType getType() {
+        return type;
+    }
+
+    @Override
+    public int hashCode() {
```

Comment:
nit: `equals` and `hashcode` can use the new `Objects.equals` and `Objects.hashcode` respectively. this is not significant issue that needs to be addressed now